glean-core/src/common_metric_data.rs

```diff
@@ -72,7 +75,52 @@ pub struct CommonMetricData {
     /// metric to be recorded to, dynamic labels are stored in the specific
     /// label so that we can validate them when the Glean singleton is
     /// available.
-    pub dynamic_label: Option<String>,
+    pub dynamic_label: Option<DynamicLabelType>,
+}
+
+// TODO FIX ERRORS. Similar errors see for all variants, not sure how to specify this in UniFFI idl:
+/*
+error[E0769]: tuple variant `Self::Label` written as struct variant
+   --> <redacted>/glean/target/debug/build/glean-core-7b31c5607feb2bbe/out/glean.uniffi.rs:125:1
+    |
+125 | #[::uniffi::udl_derive(Enum)]
+    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    |
+    = note: this error originates in the attribute macro `::uniffi::udl_derive` (in Nightly builds, run with -Z macro-backtrace for more info)
+
+*/
+
+#[derive(Debug, Clone, Deserialize, Serialize, MallocSizeOf)]
+pub enum DynamicLabelType {
+    Label(String),
+    KeyOnly(String),
+    CategoryOnly(String),
+    KeyAndCategory(String),
+}
```

Comment:
That's because in the UDL you have it defined as

    enum DynamicLabelType {
        "Label",
        "KeyAndCategory",
    };

which is not the same.
What we _really_ need is not expressible in UDL though, so we can do this in UDL:

```
typedef enum DynamicLabelType;
```

Then the following should tell UniFFI about it:

```rust
#[derive(Debug, Clone, Deserialize, Serialize, MallocSizeOf, uniffi::Enum)]
pub enum DynamicLabelType {
    Label(String),
    KeyOnly(String),
    CategoryOnly(String),
    KeyAndCategory(String),
}
```